### PR TITLE
Add promtail role to ansible-roles

### PR DESCRIPTION
**Pull request by pmaslana**

### CODE REVIEW

File: promtail/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+      - gnupg
+      - python3-apt
+    state: present
+    update_cache: true
+    cache_valid_time: 300
+  register: apt_action
+  retries: 100
+  until: apt_action is success or ('Failed to lock apt for exclusive operation' not in apt_action.msg and '/var/lib/dpkg/lock' not in apt_action.msg)
+
+- name: Create the apt keyrings directory
```
Comment (cmmarslender) [inline promtail/tasks/main.yml:17]: Dont know if it matters too much, but we use a different path for the chia keychain - might be nice to be consistent https://github.com/Chia-Network/ansible-roles/blob/main/chia-apt-key/tasks/main.yml#L21

File: promtail/tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+  ansible.builtin.get_url:
+    url: https://apt.grafana.com/gpg.key
+    dest: /etc/apt/keyrings/grafana.gpg
+    mode: '0644'
+
+- name: Convert GPG key to the required format
+  become: true
+  ansible.builtin.command: gpg --dearmor -o /etc/apt/keyrings/grafana.gpg /etc/apt/keyrings/grafana.gpg
+  args:
+    creates: /etc/apt/keyrings/grafana.gpg
```
Comment (cmmarslender) [inline promtail/tasks/main.yml:35]: I think you can replace this all with the `ansible.builtin.apt_key:` module
See: https://github.com/Chia-Network/ansible-roles/blob/main/chia-apt-key/tasks/main.yml#L18C3-L18C27

**Review by cmmarslender — CHANGES_REQUESTED** [review]

File: promtail/tasks/main.yml (hunk 1 above)
Comment (pmaslana) [reply promtail/tasks/main.yml:17]: changed the path

File: promtail/tasks/main.yml (hunk 2 above)
Comment (pmaslana) [reply promtail/tasks/main.yml]: Swapped over to use ansible.builtin.apt_key

**Review by cmmarslender — APPROVED** [review]

**Pull request merged**